File: hellogo/pkg/chess/common/perft_type.go
package chesscommon

// PerftType represents the type of Perft calculation.
// Please note that since the move generator generates only legal moves, both types should yield the same result.
type PerftType int

const (
	// NonBulk is a non-bulk Perft (Performance Test) calculation; moves at last depth are not played.
	NonBulk PerftType = iota
	// Bulk is a bulk Perft (Performance Test) calculation.
	Bulk
)

File: hellojava/src/main/java/com/fathzer/chess/common/PerftType.java
package com.fathzer.chess.common;

/** The type of Perft calculation.
 * <br>Please note that as Calvin move generator generates only legal moves both type should yield the same result.
 */
public enum PerftType {
    /** A non bulk Perft (Performance Test) calculation; moves at last depth are not played  */
    NON_BULK,
    /** A bulk Perft (Performance Test) calculation. */
    BULK
}